### PR TITLE
feat(TFD-1860) Support several criterias in FilterRow

### PR DESCRIPTION
**Pull request by lbourgeois**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/911/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/911/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/912/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/912/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/922/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/922/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/925/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/925/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/926/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/926/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/928/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/928/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/929/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/929/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/934/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/934/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/936/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/936/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/937/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/937/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/940/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/940/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/941/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/941/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/942/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/942/)

File: components/components-processing/processing-definition/src/main/java/org/talend/components/processing/definition/filterrow/FilterRowCriteriaProperties.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            @Override
+            public void afterSchema() {
+                updateFuntionColumn();
+                updateOperatorColumn();
+            }
+
+        };
+    }
+
+    public void updateFuntionColumn() {
```
Comment (RyanSkraba) [inline components/components-processing/processing-definition/src/main/java/org/talend/components/processing/definition/filterrow/FilterRowCriteriaProperties.java]: updateFunctionColumn

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/946/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/946/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/947/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/947/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/948/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/948/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/949/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/949/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/951/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/951/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/952/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/952/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/954/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/954/)

File: components/components-processing/processing-definition/src/test/java/org/talend/components/processing/definition/filterrow/FilterRowPropertiesTest.java [hunk 2]
```diff
@@ -554,17 +555,25 @@ public void testUpdateConditions_booleanSchema() {
...
+
+    @Test
+    public void testGeneratedJson() throws URISyntaxException, IOException {
+        String expectedJson = FilterRowPropertiesTest.readJson("FilterRowProperties.json");
         FilterRowProperties properties = new FilterRowProperties("test");
         properties.init();
         String jsonValue = JsonSchemaUtil.toJson(properties, Form.MAIN, FilterRowDefinition.COMPONENT_NAME);
-        // JsonSchemaUtilTest.writeJson(jsonValue, "/tmp/FilterRowProperties.json");
-        // Assert.assertEquals(expectedJson, jsonValue);
+        // JsonSchemaUtilTest.writeJson(jsonValue, "/home/lbourgeois/generated/FilterRowProperties.json");
```
Comment (RyanSkraba) [inline components/components-processing/processing-definition/src/test/java/org/talend/components/processing/definition/filterrow/FilterRowPropertiesTest.java]: Problably should delete username :D

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/955/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/955/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/957/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/957/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/958/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/958/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/959/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/959/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/960/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/960/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/961/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/961/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/962/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/962/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/966/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/966/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/967/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/967/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/977/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/977/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/980/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/980/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/981/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/981/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/982/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/982/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/983/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/983/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/986/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/986/)

**Pull request merged**